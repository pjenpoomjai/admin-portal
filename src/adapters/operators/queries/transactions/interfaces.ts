export interface ITransactionResponse {
	id: string
	service: string
	transactionEndtoEndId: string
	transactionReference: string
	transactionReceiptNumber: string
	classification: string
	creditPartyBankCode: string
	debitPartyBankCode: string
	amount: number
	currencyCode: string
	status: string
	creationDateTime: string
	debitAccountNumber: string
	isWhiteList: boolean
}

export interface ITransactionDataResponse {
	data: ITransactionResponse[]
	pageNo: number
	pageLimit: number
	totalPage: number
	totalRow: number
}

export interface ITransactionDetailPayerPayeeResponse {
	proxyCustomerType: string
	proxyAccountName: string
	availableBalance: number
	accountStatusCode: string
	accountStatusDescription: string
	emailAddress: string
	mobileNumber: string
	tepaCode: string
	fcdAmount: number
	fcdCurrency: string
	fcdEquivalentAmount: number
}

export interface ITransactionHistoryDetailResponse {
	from: string
	to: string
	event: string
	logDatetime: string
	resource?: string
	externalCode?: string
	externalDescription?: any
}

export interface ITransactionHistoryResponse {
	id: string
	referenceId: string
	logType: 'TRANSFER' | 'BILLPAYMENT'
	flow: 'REVIEW' | 'CONFIRM'
	logDetail: ITransactionHistoryDetailResponse[]
}

export interface ITransactionDetailAdditionalDetailResponse {
	creationDateTime: string
	onUs: boolean
	paymentMethod: string
	chargeAmount: number
	ibFee: number
	irFee: number
	expiryDateTime: string
	postedDateTime: string
	additionalNote: string
	status: string
	billerType: string
	billerCode: string
	billerName: string
	billerProfileId: string
	billerSubType: string
	dueDate: string
	billReference1: string
	billReference2: string
	billReference3: string
	remittanceInfo: string
	thaiQrTag: string
	paymentType: string
	senderName: string
	senderId: string
	senderMobile: string
	senderEmail: string
	transactionType: string
	annotation: string
	accountServiceReference: string
	transactionReceiptNumber: string
	equivalentAmount: number
	equivalentCurrency: string
	exchangeRate: number
	payer: ITransactionDetailPayerPayeeResponse
	payee: ITransactionDetailPayerPayeeResponse
}

export interface ITransactionDetailAddressResponse {
	addressLine1: string
	addressLine2: string
	addressLine3: string
	city: string
	country: string
	postalCode: string
}

export interface ITransactionDetailDebitCreditPartyResponse
	extends ITransactionDetailAddressResponse {
	Id: string
	id2Type: string
	account: string
	name: string
}

export interface ITransactionDetailDebitCreditPartyAgentResponse
	extends Omit<ITransactionDetailDebitCreditPartyResponse, 'id2Type'> {
	baseOn20022: string
	idType: string
}

export interface ITransactionDetailIncomingOutgoingResponse {
	instructionId: string
	businessParticipant: string
	scheme: string
}

export interface ITransactionDetailResponse {
	transactionId: string
	instructionIdentification: string
	instructionReference: string
	service: string
	transactionEndToEndId: string
	transactionReference: string
	transactionReceiptNumber: string
	classification: string
	amount: number
	currencyCode: string
	transactionDate: string
	status: string
	activityDomain: string
	linkedTransaction: string[]
	lastedSubmission: string
	servicerTransactionId: string
	bank: string
	incomingBankEntity: string
	outgoingBankEntity: string
	importance: string
	debitAccountNumber: string
	iso20022Flag: boolean
	incoming: ITransactionDetailIncomingOutgoingResponse
	outgoing: ITransactionDetailIncomingOutgoingResponse
	debitParty: ITransactionDetailDebitCreditPartyResponse
	creditParty: ITransactionDetailDebitCreditPartyResponse
	debitPartyAgent: ITransactionDetailDebitCreditPartyAgentResponse
	creditPartyAgent: ITransactionDetailDebitCreditPartyAgentResponse
	// intermediaryAgent: ITransactionDetailDebitCreditPartyAgentResponse
	additionalDetail: ITransactionDetailAdditionalDetailResponse
}
