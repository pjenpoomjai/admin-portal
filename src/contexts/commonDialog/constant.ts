import noop from 'lodash/noop'
import { TCommonDialogState } from './interface'

export const initialState: TCommonDialogState = [
	{ open: false, id: '' },
	{ openCommonDialog: noop as any, closeCommonDialog: noop as any },
]
