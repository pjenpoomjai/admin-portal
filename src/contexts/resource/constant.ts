import noop from 'lodash/noop'
import { IResourcesState } from './interface'
import { Resource } from './model'

export const initialState: IResourcesState = [
	{ resources: new Resource() },
	{ updateResource: noop as any, updateState: noop as any },
]
