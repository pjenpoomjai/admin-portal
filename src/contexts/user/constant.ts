import noop from 'lodash/noop'
import { TUserState } from './interface'

export const initialState: TUserState = [
	{ isAuthenticated: false },
	{ updateValue: noop as any },
]
