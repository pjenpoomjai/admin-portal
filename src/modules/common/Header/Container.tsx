import HeaderNavbar from 'components/template/HeaderNavbar/HeaderNavbar'
import { FC } from 'react'
import { COMPONENT_TYPE } from 'shared/componentType'
import useId from 'utils/useId'
import { IHeader, IHeaderRenderProps } from './interface'

const Container: FC<IHeader> = ({
	onLogoutHandler,
	displayName,
	lastLogon,
}: IHeaderRenderProps) => {
	return (
		<div {...useId(COMPONENT_TYPE.DIV)('root-header')}>
			<HeaderNavbar
				lastLogon={lastLogon}
				displayName={displayName}
				onLogout={onLogoutHandler}
			/>
		</div>
	)
}

export default Container
