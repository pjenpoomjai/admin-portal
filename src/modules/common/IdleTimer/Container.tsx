import { FC } from 'react'
import { IIdletimerProps } from './interface'

const Container: FC<IIdletimerProps> = () => {
	return null
}

export default Container
