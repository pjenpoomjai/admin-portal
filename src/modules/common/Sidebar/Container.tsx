import Sider from 'components/template/Sider/Sider'
import { FC } from 'react'
import { ISidebar } from './interface'

const Container: FC<ISidebar> = (props) => {
	const {
		menus,
		isOpen,
		getIsActive,
		getIsExpanded,
		handleOnClickMenu,
		handleOnClickOpen,
	} = props
	return (
		<Sider
			menus={menus}
			handleOnClickMenu={handleOnClickMenu}
			isOpen={isOpen}
			getIsActive={getIsActive}
			getIsExpanded={getIsExpanded}
			handleOnClickOpen={handleOnClickOpen}
		/>
	)
}

export default Container
