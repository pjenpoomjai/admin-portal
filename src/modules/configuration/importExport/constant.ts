import { ITab } from 'components/common/Tab'

export enum ConfigurationTab {
	DOWNLOAD = 'download',
	UPLOAD = 'upload',
}

export const configurationTabs: ITab[] = [
	{
		id: ConfigurationTab.DOWNLOAD,
		name: 'Download Configuration',
		value: ConfigurationTab.DOWNLOAD,
	},
	{
		id: ConfigurationTab.UPLOAD,
		name: 'Upload Configuration',
		value: ConfigurationTab.UPLOAD,
	},
]
