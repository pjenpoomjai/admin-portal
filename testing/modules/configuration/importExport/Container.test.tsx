import { render } from '@testing-library/react'
import Container from 'modules/configuration/importExport/Container'
import { ConfigurationTab } from 'modules/configuration/importExport/constant'
import { useForm } from 'react-hook-form'
import { QueryKey } from 'shared/adapters/interfaces'
import { buildMockUseAppContext } from 'testing/test-utils/buildMockUseAppContext'
import { useAppContext } from 'x-core-modules/builder'

jest.mock('x-core-modules/builder', () => ({
	...jest.requireActual('x-core-modules/builder'),
	useAppContext: jest.fn(),
}))

const mockUseAppContext = useAppContext as jest.Mock

const mockPostImport = jest.fn()
const mockPostExport = jest.fn()

describe('Configuration Import Export : Container', () => {
	afterEach(() => {
		jest.resetModules()
		jest.clearAllMocks()
	})

	test('Should render successfully initiate state (export tab)', () => {
		// mock state
		const getQueryOperator = jest.fn().mockReturnValue({
			[QueryKey.Config]: {
				configurationImport: mockPostImport,
				configurationExport: mockPostExport,
			},
		})
		const mockAppContext = buildMockUseAppContext({
			linkAdapterCache: {
				getQueryOperator,
			},
		})
		mockUseAppContext.mockReturnValue(mockAppContext)
		const { baseElement } = render(<div></div>, {
			wrapper: () => {
				// eslint-disable-next-line react-hooks/rules-of-hooks
				const { control } = useForm<any>()

				return (
					<Container
						{...mockAppContext}
						selectedTab={ConfigurationTab.DOWNLOAD}
						tabHandler={jest.fn()}
						downloadForm={{
							onSubmitDownload: jest.fn(),
						}}
						uploadForm={{
							processImport: undefined,
							processFailDetailImport: undefined,
							isDisableUpload: true,
							onSubmitUpload: jest.fn(),
							control: control,
							onRemoveFile: jest.fn(),
						}}
					/>
				)
			},
		})
		expect(baseElement).toMatchSnapshot()
	})

	test('Should render successfully initiate state (import tab)', () => {
		// mock state
		const getQueryOperator = jest.fn().mockReturnValue({
			[QueryKey.Config]: {
				configurationImport: mockPostImport,
				configurationExport: mockPostExport,
			},
		})
		const mockAppContext = buildMockUseAppContext({
			linkAdapterCache: {
				getQueryOperator,
			},
		})
		mockUseAppContext.mockReturnValue(mockAppContext)
		const { baseElement } = render(<div></div>, {
			wrapper: () => {
				// eslint-disable-next-line react-hooks/rules-of-hooks
				const { control } = useForm<any>()

				return (
					<Container
						{...mockAppContext}
						selectedTab={ConfigurationTab.UPLOAD}
						tabHandler={jest.fn()}
						downloadForm={{
							onSubmitDownload: jest.fn(),
						}}
						uploadForm={{
							processImport: undefined,
							processFailDetailImport: undefined,
							isDisableUpload: true,
							onSubmitUpload: jest.fn(),
							control: control,
							onRemoveFile: jest.fn(),
						}}
					/>
				)
			},
		})
		expect(baseElement).toMatchSnapshot()
	})
})
