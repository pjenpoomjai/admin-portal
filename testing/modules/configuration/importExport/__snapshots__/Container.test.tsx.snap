// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration Import Export : Container Should render successfully initiate state (export tab) 1`] = `
.emotion-1 {
  overflow: hidden;
  min-height: 48px;
  -webkit-overflow-scrolling: touch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:599.95px) {
  .emotion-1 .MuiTabs-scrollButtons {
    display: none;
  }
}

.emotion-2 {
  position: relative;
  display: inline-block;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  white-space: nowrap;
  overflow-x: hidden;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.25;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  max-width: 360px;
  min-width: 90px;
  position: relative;
  min-height: 48px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 12px 16px;
  overflow: hidden;
  white-space: normal;
  text-align: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4.Mui-selected {
  color: #1976d2;
}

.emotion-4.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  position: absolute;
  height: 2px;
  bottom: 0;
  width: 100%;
  -webkit-transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #1976d2;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  box-shadow: unset;
}

.emotion-11::-moz-focus-inner {
  border-style: none;
}

.emotion-11.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-11 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-11:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-11:hover {
    background-color: #1976d2;
  }
}

.emotion-11:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-11.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-11.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-11.MuiButton-root {
  text-transform: none;
  color: var(--gray-30);
  border-radius: 6px;
  background-color: var(--violet-500);
}

.emotion-11:hover {
  background-color: var(--violet-700);
}

.emotion-11:disabled {
  background-color: var(--gray-300);
}

.emotion-11:focus {
  border: 3px solid var(--violet-300);
}

<body>
  <div>
    <div
      class="root w-full MuiBox-root emotion-0"
    >
      <p
        class="block antialiased font-sans text-xl font-semibold text-indigo-500 pb-6"
        data-id="config-typography"
        data-testid="config-typography"
        placeholder=""
      >
        Configuration
      </p>
      <div
        class="tab"
      >
        <div
          class="flex flex-row bg-white h-full overflow-y-hidden"
        >
          <div
            class="MuiTabs-root mx-[24px] emotion-1"
            data-id="transaction-tab"
            data-testid="transaction-tab"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed emotion-2"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                class="MuiTabs-flexContainer emotion-3"
                role="tablist"
              >
                <button
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected overflow-y-hidden text-gray-500 emotion-4"
                  data-id="download-tab"
                  data-testid="download-tab"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="overflow-y-hidden"
                  >
                    <p
                      class="text-base font-medium"
                      data-id="label-download-label"
                      data-testid="label-download-label"
                    >
                      Download Configuration
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root emotion-5"
                  />
                </button>
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary overflow-y-hidden text-gray-500 emotion-4"
                  data-id="upload-tab"
                  data-testid="upload-tab"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <div
                    class="overflow-y-hidden"
                  >
                    <p
                      class="text-base font-medium"
                      data-id="label-upload-label"
                      data-testid="label-upload-label"
                    >
                      Upload Configuration
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root emotion-5"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator emotion-8"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="my-[8px] h-[40px] flex items-center"
          />
        </div>
      </div>
      <div
        class="bg-white p-6 rounded-b-xl MuiBox-root emotion-0"
      >
        <div
          class="flex justify-center MuiBox-root emotion-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary text-sm font-semibold h-[34px] emotion-11"
            data-id="download-config-button"
            data-testid="download-config-button"
            id="download-config"
            tabindex="0"
            type="button"
          >
            Download
            <span
              class="MuiTouchRipple-root emotion-5"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Configuration Import Export : Container Should render successfully initiate state (import tab) 1`] = `
.emotion-1 {
  overflow: hidden;
  min-height: 48px;
  -webkit-overflow-scrolling: touch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:599.95px) {
  .emotion-1 .MuiTabs-scrollButtons {
    display: none;
  }
}

.emotion-2 {
  position: relative;
  display: inline-block;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  white-space: nowrap;
  overflow-x: hidden;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.25;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  max-width: 360px;
  min-width: 90px;
  position: relative;
  min-height: 48px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 12px 16px;
  overflow: hidden;
  white-space: normal;
  text-align: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4.Mui-selected {
  color: #1976d2;
}

.emotion-4.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  position: absolute;
  height: 2px;
  bottom: 0;
  width: 100%;
  -webkit-transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #1976d2;
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  box-shadow: unset;
}

.emotion-12::-moz-focus-inner {
  border-style: none;
}

.emotion-12.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-12 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-12:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-12:hover {
    background-color: #1976d2;
  }
}

.emotion-12:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-12.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-12.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-12.MuiButton-root {
  text-transform: none;
  color: var(--gray-30);
  border-radius: 6px;
  background-color: var(--violet-500);
}

.emotion-12:hover {
  background-color: var(--violet-700);
}

.emotion-12:disabled {
  background-color: var(--gray-300);
}

.emotion-12:focus {
  border: 3px solid var(--violet-300);
}

<body>
  <div>
    <div
      class="root w-full MuiBox-root emotion-0"
    >
      <p
        class="block antialiased font-sans text-xl font-semibold text-indigo-500 pb-6"
        data-id="config-typography"
        data-testid="config-typography"
        placeholder=""
      >
        Configuration
      </p>
      <div
        class="tab"
      >
        <div
          class="flex flex-row bg-white h-full overflow-y-hidden"
        >
          <div
            class="MuiTabs-root mx-[24px] emotion-1"
            data-id="transaction-tab"
            data-testid="transaction-tab"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed emotion-2"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                class="MuiTabs-flexContainer emotion-3"
                role="tablist"
              >
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary overflow-y-hidden text-gray-500 emotion-4"
                  data-id="download-tab"
                  data-testid="download-tab"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <div
                    class="overflow-y-hidden"
                  >
                    <p
                      class="text-base font-medium"
                      data-id="label-download-label"
                      data-testid="label-download-label"
                    >
                      Download Configuration
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root emotion-5"
                  />
                </button>
                <button
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected overflow-y-hidden text-gray-500 emotion-4"
                  data-id="upload-tab"
                  data-testid="upload-tab"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="overflow-y-hidden"
                  >
                    <p
                      class="text-base font-medium"
                      data-id="label-upload-label"
                      data-testid="label-upload-label"
                    >
                      Upload Configuration
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root emotion-5"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator emotion-8"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="my-[8px] h-[40px] flex items-center"
          />
        </div>
      </div>
      <div
        class="bg-white p-6 rounded-b-xl MuiBox-root emotion-0"
      >
        <div
          class="flex justify-center items-center flex-col MuiBox-root emotion-0"
        >
          <div
            class="flex flex-col"
          >
            <div
              class="flex mb-[0px]"
            >
              <input
                accept=".xls, .xlsx"
                class="hidden"
                data-id="input-file-upload-file-config-upload-file"
                data-testid="input-file-upload-file-config-upload-file"
                id="input-file-upload-file-config"
                name="input-file-upload-file-config"
                type="file"
              />
              <label
                class="w-[113px] h-[40px] rounded-[6px] cursor-pointer flex flex-row justify-center items-center border-gray-300 border-[1px]"
                data-id="upload-file-upload-file-config-upload-file"
                data-testid="upload-file-upload-file-config-upload-file"
              >
                <span
                  class="text-sm font-semibold"
                >
                  Select file
                </span>
              </label>
              <div
                class="text-xs text-gray-600 ml-[16px] my-[12px]"
                data-id="upload-file-desc-upload-file-config-div"
                data-testid="upload-file-desc-upload-file-config-div"
              >
                Supported file types: XLS or XLSX up to 10MB
              </div>
            </div>
          </div>
          <p
            class="block antialiased font-sans font-light text-sm text-gray-600 pt-4"
            data-id="make-sure-upload-typography"
            data-testid="make-sure-upload-typography"
            placeholder=""
          >
            Make sure the file is from the Downloads tab before you upload.
          </p>
        </div>
        <div
          class="flex justify-center mt-4 MuiBox-root emotion-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary emotion-12"
            data-id="upload-config-button"
            data-testid="upload-config-button"
            disabled=""
            id="upload-config"
            tabindex="-1"
            type="button"
          >
            Upload
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
